crates/ruff/src/rules/flake8_tidy_imports/relative_imports.rs

```diff
@@ -99,19 +99,29 @@ fn fix_banned_relative_import(
             parts.pop();
         }
 
-        let call_path = from_relative_import(&parts, module.unwrap());
-        let module_name = call_path.as_slice();
+        let module_name = if let Some(module) = module {
+            let call_path = from_relative_import(&parts, module);
+            call_path.as_slice().join(".")
+        } else {
+            if parts.len() > 1 {
+                let module = parts.last().unwrap();
+                let call_path = from_relative_import(&parts, module);
+                call_path.as_slice().join(".")
+            } else {
+                parts.join(".")
```

Comment:
@charliermarsh Here I wanted to return `parts` as slice, but couldn't get the types to the other arms to match (`Vec<String>` vs `Vec<&str>`). Any pointers? Still learning idiomatic rust..